File: include/correlations/pseudo-reduced-ratios.rb

begin_correlation("Tpr", "GasPseudoreducedTemperature", "PseudoReducedTemperature") 
add_title("CALCULATION OF THE PSEUDOREDUCED TEMPERATURE")
add_ref("ahmed:1989")
add_ref("banzer:1996")
add_parameter("t",  "Rankine", "Temperature")
add_parameter("tpc",  "Rankine", "Gas pseudocritical temperature")
add_author("Standard Equation")
end_correlation()

################################################################

begin_correlation("Ppr", "GasPseudoreducedPressure", "PseudoReducedPressure") 
add_title("CALCULATION OF THE PSEUDOREDUCED PRESSURE")
add_ref("ahmed:1989")
add_ref("banzer:1996")
add_parameter("p",  "psia", "Pressure")
add_parameter("ppc",  "psia", "Gas pseudocritical pressure")
add_author("Standard Equation")
end_correlation()

File: include/correlations/pseudo-reduced-ratios.cc
Quantity<PseudoReducedTemperature> Tpr__correlation__fct(
    const Quantity<Rankine>& t, const Quantity<Rankine>& tpc) {
  return Tpr::get_instance()(t, tpc);
}

double Tpr__correlation__valid__fct(const double& t, const double& tpc) {
  return Tpr::get_instance()(t, tpc).raw();
}

Quantity<PseudoReducedPressure> Ppr__correlation__fct(
    const Quantity<psia>& p, const Quantity<psia>& ppc) {
  return Ppr::get_instance()(p, ppc);
}

double Ppr__correlation__valid__fct(const double& p, const double& ppc) {
  return Ppr::get_instance()(p, ppc).raw();
}
